### PR TITLE
Convert Dates List Table View to TypeScript; closes #2450.

### LINKED ISSUE #2450: Convert Dates List Table View to TypeScript
- see `domain/eventEditor/ui/datetimes/datesList/tableView`
- also convert to use Apollo GQL (AGQL) entity types if not done so already

### PR DESCRIPTION
**Pull request by nicolad**
## Problem this Pull Request solves
- Closes https://github.com/eventespresso/event-espresso-core/issues/2450
- Also adds `breakpoint` mixin, but that is optional and might be reverted.

### CODE REVIEW

File: assets/src/domain/eventEditor/ui/datetimes/datesList/tableView/styles.scss [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+			@media (min-width: $breakpoint) {
+				@content;
+			}
+		} @else {
+			@media ($direction: $breakpoint) {
+				@content;
+			}
+		}
+	}
+}
```
Comment (nicolad) [inline assets/src/domain/eventEditor/ui/datetimes/datesList/tableView/styles.scss]: source of the mixin: https://rimdev.io/making-media-query-mixins-with-sass/

File: assets/src/domain/eventEditor/ui/datetimes/datesList/tableView/TableView.tsx [hunk 2]
```diff
@@ -7,29 +7,23 @@ import datesListTableHeader from './datesListTableHeader';
 import datesListTableRow from './datesListTableRow';
 import useDatetimeItem from '@edtrServices/apollo/queries/datetimes/useDatetimeItem';
 
-import './editor-date-entities-list-view.css';
+import { TableViewProps } from './types';
+
+import './styles.scss';
```
Comment (nicolad) [inline assets/src/domain/eventEditor/ui/datetimes/datesList/tableView/TableView.tsx]: thought that `editor-date-entities-list-view` is a bit mouthful and renames

File: assets/src/domain/eventEditor/ui/datetimes/datesList/tableView/styles.scss [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+.ee-editor-date-actions-menu {
+	@include breakpoint(mobile, max) {
+		justify-content: center;
+	}
+	@include breakpoint(tablet, max) {
+		justify-content: start;
+	}
```
Comment (nicolad) [inline assets/src/domain/eventEditor/ui/datetimes/datesList/tableView/styles.scss]: this is the main benefit/ use case of the breakpoint

File: assets/src/domain/eventEditor/ui/datetimes/datesList/tableView/styles.scss (hunk 1 above)
Comment (nicolad) [reply assets/src/domain/eventEditor/ui/datetimes/datesList/tableView/styles.scss]: @tn3rb if you think we might need this, please suggest and appropriate location

File: assets/src/domain/eventEditor/ui/datetimes/datesList/tableView/TableView.tsx [hunk 4]
```diff
@@ -7,29 +7,23 @@ import datesListTableHeader from './datesListTableHeader';
...
-	className,
-	displayDates,
-	entities: datetimes,
-	allEventDates,
-	showDate,
-	setEntityIds,
-	setSortBy,
-	...props
-}) => {
+const TableView: React.FC<TableViewProps> = ({ className, displayDates, entities: datetimes, showDate, ...props }) => {
```
Comment (manzoorwanijk) [inline assets/src/domain/eventEditor/ui/datetimes/datesList/tableView/TableView.tsx]: May be we should rename `displayDates` and `showDate` to some more explicit and clear names, that can be done in another PR.

**Comment by nicolad** [comment]
the result: 
image

**Comment by nicolad** [comment]
result: 
image

**Comment by nicolad** [comment]
output with that 782px specific media query:
image

File: assets/src/domain/eventEditor/ui/datetimes/datesList/tableView/styles.scss (hunk 1 above)
Comment (nicolad) [reply assets/src/domain/eventEditor/ui/datetimes/datesList/tableView/styles.scss]: Moved into mixins folder, in suggested place.
I noticed the same approach here: https://github.com/twbs/bootstrap/tree/master/scss/mixins

File: assets/src/domain/eventEditor/ui/datetimes/datesList/tableView/TableView.tsx (hunk 4 above)
Comment (nicolad) [reply assets/src/domain/eventEditor/ui/datetimes/datesList/tableView/TableView.tsx]: agree with this suggestion :+1:

**Pull request merged**